[EDU-6696] API V4 |  MIGRATION REFERENCE (#1766)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/secure/api-v4-migration.mdx

- The migration affects both the platform (with the introduction of Workloads and Edge Connector) and the API endpoints.
- Some configuration settings have moved to new locations to improve reusability and clarity.
- New personal accounts created from **August 1, 2025**, will use API v4 by default. Existing accounts will be migrated in waves; see the migration schedule below for details.
- Azion will contact customers with a Technical Account Manager contract to coordinate the migration. Developer-tier accounts will migrate automatically.
- During the transition, you may see documentation with banners allowing you to switch between v3 and v4 versions of a page.

API v4 represents a technological shift, providing a foundation for future features and improved operational efficiency across the Azion Web Platform.

---

## API migration

All API endpoints will be migrated to v4 in waves. To view the available endpoints and parameters, see the [API v4 reference](api.azion.com/v4)

API v3 is now **deprecated** and will no longer receive updates. It will be discontinued on **March 1, 2026**. After this date, all operations must be performed exclusively using API v4.

> ### Required action

Reviewer: Não é melhor 39-41 ser uma tip dentro de API migration (33-37)?

### Operational Changes and User-Facing Impact

The following settings have moved from **Azion Domains** to  **Workloads**.

| Setting                        | Previous Location      | New Location in v4         |
|---------------------------------|-----------------------|----------------------------------------|
| Environment Type                | Domains               | Workload (top-level)                   |
| Domain Settings                 | Domains               | Workload > Domain Settings             |
| Edge Application                | Domains > Settings    | Workload > Deployment Settings         |
| Edge Firewall                   | Domains > Settings    | Workload > Deployment Settings         |
| CNAME                           | Domains > Settings    | Workload > Domain Settings             |
| Digital Certificate             | Domains > Settings    | Workload > Protocol Settings           |
| CNAME Access Only               | Domains > Settings    | Workload > Domain Settings             |
| mTLS (Mutual TLS) Settings      | Domains               | Workload > Domain Settings             |

> **Delivery Settings** previously configured within Edge Application, now set in **Workloads** under **Protocol Settings**:

Reviewer: ```suggestion
**Delivery Settings** previously configured within an Edge Application can now be found in **Workloads** under **Protocol Settings**:
```

- Protocol Settings
- Delivery Ports
- Minimum TLS version
- TLS Ciphers

#### Notable Behavioral Changes

> - **CNAME Access Only**

Reviewer: A indentação aqui difere de 100 e 101, interfere?

>   - *Previous behavior*: When enabled, the application could only be accessed via CNAME (e.g., `app.example.com`).

Reviewer: ```suggestion
  - *Previous behavior*: When enabled, the application could only be accessed via CNAME (for example, `app.example.com`).
```

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Example Workload",
  "active": true,
  "infrastructure": 1,
  "protocols": {
    "http": {
      "versions": [
        "http1",
        "http2"
      ]
    }
  },
  "workload_domain_allow_access": true
}'
`} />

**Creating a Workload Deployment**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
    }
  }
}'
`} />

:::tip
For more information about the API parameters visit the [Workloads API Reference](https://api.azion.com/v4#/operations/GetWorkspaceWorkloads)
:::

### What happens to my existing Domains?

> Domains will be automatically converted to Workloads in phases, ensuring compatibility with the original configurations. The migration will occur in waves. For more information about migration dates, see the details provided below.

Reviewer: Suggestion: The migration will happen in phases. For detailed information about the schedule, please refer to the details below.

> After your account is migrated youll will not be able to create new Domains.

Reviewer: ```suggestion
After your account is migrated you'll not be able to create new Domains.
```

### Required action

If you have an automation that currently uses the **Domains** API Endpoints you must update it to use v4 Workloads endpoints.

### New Features

> - Introduction of [Custom Pages](#custom-pages), which replaces and expands on the previous “Error Responses” feature. More details are provided below.

Reviewer: O link #custom-pages tá completo?

- Azion Custom Domain allows you to set up a custom domain for your application using the format `example.azion.app`.

### Migration Benefits

- **Improved Organization**: Logically group all resources related to an application or service into a single unit (Workload), making management, visualization, and navigation more efficient. This structure creates clear relationships between components.
- **Scalability and Flexibility**: Simplifies deployment, updates, and scaling of complex applications by allowing all required components to be managed as part of a Workload.
- **Simplified Management**: Centralizes configuration and control of different application elements, reducing operational complexity.
- **Foundation for New Capabilities**: Required for future features such as configuration versioning and more intuitive service connections, for example, linking a domain directly to Edge Storage without relying on an Edge Application.

> <LinkButton link="/documentation/products/secure/workloads" text="Visit the Workloads Reference" />

Reviewer: Falta /en aqui /documentation/products/secure/workloads?

---

## Edge Connector

**Edge Connector** centralizes connection settings, allowing you to reuse them across multiple applications, and provides granular control over performance, security, and routing through a unified interface for connecting **Edge Applications** to an origin (your data source), natively incorporating features such as load balancing, HMAC authentication, and Origin Shield.

> **Edge Connector*** replaces Edge Appllication's **Origin** settings. All settings previously managed in Edge Aplications **Origins** tab are now organized under **Edge Connector***.

Reviewer: ```suggestion
**Edge Connector*** replaces Edge Application's **Origin** settings. All settings previously managed in Edge Aplications **Origins** tab are now organized under **Edge Connector***.
```

### Operational Changes and User-Facing Impact

> All settings previously managed in Edge Aplications **Origins** tab are now organized under **Edge Connector***.

Reviewer: ```suggestion
All settings previously managed in Edge Applications **Origins** tab are now organized under **Edge Connector***.
```

  - *New behavior*: Single Origin and Load Balancer have been unified into the `HTTP` type. When using multiple origins, you must enable the **Load Balancer Module** to add more origins. 

- **HMAC Authentication**
  - *Previous behavior*: Availiable when configuring an origin.
  - *New behavior*: Now you must enable the **Origin Shield Module** when configuring an `HTTP` type **Edge Connector**.

#### API Usage

You can manage **Edge Connectors** and their configuration through the API using the available v4 endpoints. Refer to the [API reference](https://api.azion.com/v4#/operations/GetEdgeConnectorConnectors) for details on supported operations and parameters.

##### Example requests

**Creating a Edge Connector with the Edge Storage Connector Type**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/edge_connector/connectors \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "edge storage connector",
  "active": true,
  "type": "edge_storage",
  "attributes": {
    "bucket": "app-origin",
    "prefix": "/"
}'
`} />

> **Creating a Edge Connector with the HTTP Connector Type**:

Reviewer: ```suggestion
**Creating an Edge Connector with the HTTP Connector Type**:
```

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/edge_connector/connectors \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
    "name": "example.com",
    "active": true,
    "type": "http",
    "attributes": {
      "addresses": [
        {
          "active": true,
          "address": "httpbin.org",
          "http_port": 80,
          "https_port": 443,
          "modules": null
        }
      ],
      "connection_options": {
        "dns_resolution": "preserve",
        "transport_policy": "preserve",
        "http_version_policy": "http1_1",
        "host": "httpbin.org",
        "path_prefix": "/",
        "following_redirect": true,
        "real_ip_header": "X-Real-IP",
        "real_port_header": "X-Real-PORT"
      },
      "modules": {
        "load_balancer": {
          "enabled": false,
          "config": null
        },
        "origin_shield": {
          "enabled": false,
          "config": null
        }
      }
    }
  }'
`} />

:::tip

> For more information about the API parameters visit the [Edge Conenctor API Reference](https://api.azion.com/v4#/operations/GetEdgeConnectorConnectors)

Reviewer: ```suggestion
For more information about the API parameters visit the [Edge Connector API reference](https://api.azion.com/v4#/operations/GetEdgeConnectorConnectors).
```

:::

### What happens to my existing Origins?

Existing Origins will be automatically converted to **Edge Connectors** in phases, maintaining compatibility with the original configurations. The migration will occur in waves. For more information about migration dates, see the details provided below.

### Required action

If you have an automation that currently uses the **Edge Application - Origins** API Endpoints you must update it to use v4 endpoints.

### New Features

> - **Configuration reusability**: An Edge Connector can be connected to multiple Edge Applications

Reviewer: ```suggestion
- **Configuration reusability**: An Edge Connector can be connected to multiple Edge Applications.
```

### Migration Benefits

**Simplified Backend Management**: Edge Connector centralizes all backend connection definitions, reducing configuration complexity. Multiple **Edge Applications** can use a single **Edge Connector**, streamlining maintenance and ensuring configuration consistency.

> **Future-Ready Architecture**: **Edge Connector** provides the foundation for upcoming Azion Web Platform features and optimizations, ensuring your Edge Applications are ready to leverage new capabilities as they become available.

Reviewer: ```suggestion
- **Future-Ready Architecture**: **Edge Connector** provides the foundation for upcoming Azion Web Platform features and optimizations, ensuring your Edge Applications are ready to leverage new capabilities as they become available.
```

> <LinkButton link="/documentation/products/secure/edge-connector" text="Visit the Edge Connector Reference"/>

Reviewer: Ver se adapta reference list ou guide

---

## Custom Pages

**Custom Pages** is an Azion feature that allows you to customize error pages for users based on the status code received from an [Edge Connector](/en/documentation/products/secure/edge-connector/) when fetching the content from your origin.

**Custom Pages*** replaces Edge Appllication's **Error Responses** settings. All settings previously managed in Edge Aplications **Error Responses** tab are now organized under **Custom Pages**.

### Operational Changes and User-Facing Impact

> All settings previously managed in Edge Aplications **Error Responses** tab are now organized under **Custom Pages**.

Reviewer: Mesma informação da linha 313...teria como otimizar e manter só uma vez? Sugestão: Custom Pages now replaces all settings that were previously managed under Edge Applications’ Error Responses tab.

|-----------------------------|---------------------------------------------------|---------------------------------------------------------|
| Default Error Response      | Edge Application > Error Responses                | Custom Pages > Page Codes              |
| Custom Error Response       | Edge Application > Error Responses                | Custom Pages                                            |
| Status Code                 | Edge Application > Error Responses > Custom Error | Custom Pages > Status Configuration > Page Code         |
| Page Path                   | Edge Application > Error Responses > Custom Error | Custom Pages > Response Details > Page Path (URI)       |
| Default Response TTL        | Edge Application > Error Responses > Custom Error | Custom Pages > Response Details > Response TTL          |
| Set Origin                  | Edge Application > Error Responses > Set Origin   | Custom Pages > Response Details > Page Path (URI)       |

#### Notable Behavioral Changes

- **Relationship**
  - *Previous behavior*: Error pages were a part of an **Edge Application**.

>   - *New behavior*: Custom Pages are separate resources and are linked to an **Workload**.

Reviewer: ```suggestion
  - *New behavior*: Custom Pages are separate resources and are linked to a **Workload**.
```

#### API Usage

You can manage **Custom Pages** and their configuration through the API using the available v4 endpoints. Refer to the [API reference](api.azion.com/v4#/operations/GetWorkspaceCustomPages) for details on supported operations and parameters.

> To use an **Custom Page** you must link it to an **Workload** using the **Workload deployment** Endpoint. When using the Console, this association is created under an Workload's Deployment Settings.

Reviewer: ```suggestion
To use a **Custom Page**, you must link it to a **Workload** using the **Workload deployment** endpoint. When using the Console, this association is created under a Workload's deployment setting.
```

##### Example requests

**Creating a Custom Page**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/custom_pages \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "401 error page",
  "active": true,
  "pages": [
    {
      "code": "401",
      "page": {
        "type": "page_connector",
        "attributes": {
          "connector": <edge_connector_id>,
          "ttl": 300,
          "uri": "/path/error_page.html",
          "custom_status_code": 100
        }
      }
    }
  ]
}'
`} />

> **Binding an Custom Page to an Workload Deployment**:

Reviewer: ```suggestion
**Binding a Custom Page to a Workload Deployment**:
```

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
      "custom_page": <custom_page_id>
    }
  }
}'
`} />

> 

Reviewer: ```suggestion
```

:::tip

> For more information about the API parameters visit the [Custom Pages API Reference](https://stage-api.azion.com/v4#/operations/GetWorkspaceCustomPages)

Reviewer: API Reference list? API Reference guide? API Reference page?
```suggestion
For more information about the API parameters visit the [Custom Pages API Reference](https://stage-api.azion.com/v4#/operations/GetWorkspaceCustomPages).
```

:::

### What happens to my existing Error Responses?

Existing Error Responses will be automatically converted to **Custom Pages** in phases, maintaining compatibility with the original configurations. The migration will occur in waves. For more information about migration dates, see the details provided below.

### New Features

> - **Response Details**: Customize the response for this status code. Specify the URI, status code, TTL, content type, and the response body to be returned to the client.

Reviewer: fala em "this status code" mas não menciona qual. Necessária a informação?

### Migration Benefits

<LinkButton link="/documentation/products/secure/custom-pages" text="Visit the Custom Pages Reference"/>

---

## Planned migration timeline

| Scope | Developer | Business | Enterprise | Mission Critical |
| --- | --- | --- | --- | ---|
|Planned Migration Date | 100 | 100 | 100 | 1,000 |

---

## Recommendations

Here are recommended actions to ensure a smooth migration:

- **Monitor Azion communications**: Stay informed about migration dates, guides, and any tools provided for the transition.
- **Plan your migration**: Assess how your current settings will fit into the new structure and strategically plan the migration of your resources.

> - **Prepare your team**: Ensure your team understands the Workloads concept and its impact on application development and deployment on the Azion platform.

Reviewer: ```suggestion
- **Prepare your team**: Ensure your team understands the Workloads concept and its impact on application development and deployment on the Azion Web Platform.
```

- **Test new features**: As the new features are implemented, test the new capabilities to optimize your applications and services.

---

---
## Q&A

> ### How can i test the new features?

Reviewer: ```suggestion
### How can I test the new features?
```

Starting Augst 1st 2025, new Developer accounts will have access to the new features. You can create a personal account not linked to your companys account, and have access to test the new features.

> ### Do i Have to Do Something?

Reviewer: Sugestão de pergunta mais clara: Do I need to take any action during the migration?
```suggestion
### Do I have to do something?
```

Most users will not need to take any action, as the migration process is automatic and preserves existing configurations. However, if you use custom automation, scripts, or CI/CD pipelines that interact with the previous API endpoints, you must update them to use the new v4 endpoints. Review your integrations to ensure compatibility after migration.

> ## Need Assistance?

Reviewer: ```suggestion
## What should I do if I need assistance?"
```

> You can contact our support team or ask Azion Copilot

Reviewer: ```suggestion
You can contact our support team via the [Azion Support Portal]((https://tickets.azion.com/en/support/) ) or ask Azion Copilot directly within the [Azion Console](https://console.azion.com/).
```

`} />

**Creating a Workload Deployment**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
    }
  }
}'
`} />

:::tip
For more information about the API parameters visit the [Workloads API Reference](https://api.azion.com/v4#/operations/GetWorkspaceWorkloads)
:::

### What happens to my existing Domains?

Domains will be automatically converted to Workloads in phases, ensuring compatibility with the original configurations. The migration will occur in waves. For more information about migration dates, see the details provided below.

After your account is migrated you'll not be able to create new Domains.

### Required action

If you have an automation that currently uses the **Domains** API endpoints you must update it to use v4 **Workloads** endpoints.

### New Features

> - **Custom Pages**: Replaces and expands on the previous Error Responses feature. More details are provided [below](#custom-pages).

Reviewer: ```suggestion
- **Custom Pages**: Replaces and expands on the previous Error Responses feature. [More details are provided below](#custom-pages).
```

          "https_port": 443,
          "modules": null
        }
      ],
      "connection_options": {
        "dns_resolution": "preserve",
        "transport_policy": "preserve",
        "http_version_policy": "http1_1",
        "host": "example.com",
        "path_prefix": "/",
        "following_redirect": true,
        "real_ip_header": "X-Real-IP",
        "real_port_header": "X-Real-PORT"
      },
      "modules": {
        "load_balancer": {
          "enabled": false,
          "config": null
        },
        "origin_shield": {
          "enabled": false,
          "config": null
        }
      }
    }
  }'
`} />

:::tip
For more information about the API parameters visit the [Edge Connector API reference](https://api.azion.com/v4#/operations/GetEdgeConnectorConnectors).
:::

### What happens to my existing Origins?

Existing Origins will be automatically converted to **Edge Connectors** in phases, maintaining compatibility with the original configurations. The migration will occur in waves.

### Required action

If you have an automation that currently uses the **Edge Application - Origins** API endpoints you must update it to use v4 endpoints.

### New Features

> - **Configuration reusability**: An Edge Connector can be connected to multiple Edge Applications.

Reviewer 2: PAra deixar mais claro o benefício:

De: An Edge Connector can...
Para:  The same Edge Connector can be...

        "attributes": {
          "connector": <edge_connector_id>,
          "ttl": 300,
          "uri": "/path/error_page.html",
          "custom_status_code": 100
        }
      }
    }
  ]
}'
`} />

**Binding a Custom Page to a Workload Deployment**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
      "custom_page": <custom_page_id>
    }
  }
}'
`} />

:::tip
For more information about the API parameters visit the [Custom Pages API Reference](https://api.azion.com/v4#/operations/GetWorkspaceCustomPages)
:::

### What happens to my existing Error Responses?

Existing Error Responses will be automatically converted to **Custom Pages** in phases, maintaining compatibility with the original configurations. The migration will occur in waves.

### New Features

> - **Response Details**: Customize the response for an status code. Specify the URI, status code, TTL, content type, and the response body to be returned to the client.

Reviewer: ```suggestion
- **Response Details**: Customize the response for a status code. Specify the URI, status code, TTL, content type, and the response body to be returned to the client.
```

- **Per-Status Code Edge Connector Assignment** Assign a specific **Edge Connector** to each error response page code in **Custom Pages**. This allows you to customize the response for each error code individually.

### Migration Benefits

- **Support for Multiple Status Codes**: Configure custom responses for a wide range of HTTP status codes, including `400`, `401`, `403`, `404`, and others.
- **Comprehensive Configuration Options**:
  - Define a custom status code for each page.
  - Set the response Time to Live (TTL).
  - Specify the path to the custom page.
  - Link each custom page to one or more Edge Connectors.

>   - Use the Type field to associate the custom page with different contexts, such as a specific **Edge Connector**

Reviewer: ```suggestion
  - Use the Type field to associate the custom page with different contexts, such as a specific **Edge Connector**.
```

---
title: API v4 Migration
description: >-
  Find detailed informations about Azions plan for migration to our new API v4.

> meta_tags: 'edge application, workloads, secure'

Reviewer 3: Edge Connector?

- During the transition, you may see documentation with banners allowing you to switch between v3 and v4 versions of a page.
- After your account is migrated you will not be able to use the the v3 API endpoints or create new `Domains`, `Origins` or `Error Responses`. You must use the equivalent new resources. 

API v4 represents a technological shift, providing a foundation for future features and improved operational efficiency across the Azion Web Platform.

---

## API migration

All API endpoints will be migrated to v4 in waves. To view the available endpoints and parameters, see the [API v4 reference](api.azion.com/v4)

API v3 is now **deprecated** and will no longer receive updates. It will be discontinued on **March 1, 2026**. After this date, all operations must be performed exclusively using API v4.

### Required action

Automations, scripts, and CI/CD pipelines connected to Azion must be updated to use API v4 by the specified deadline.

### New Features

- JWT authentication.
- Activity History via GraphQL for full auditability.

> - New event-driven architecture.

Reviewer 2: @LuizaVSantos: Precisamos citar outros benefícios como a introdução de novos argumentos que auxiliam muito nas operações de consulta:

search
string
A search term.

fields
string
Comma-separated list of field names to include in the response.

Fonte: [Query Parameters](https://api.azion.com/v4#/operations/GetDigitalCertificatesCrls#Query-Parameters)

<img width="1394" height="848" alt="image" src="https://github.com/user-attachments/assets/1670493e-f622-4e7d-b8a9-3292eca18290" />

### Migration Benefits

- Improved performance
- Increased flexibility

---

## Changes in Edge Applications

The functionality of Edge Applications remains unchanged. However, some configuration options that were previously managed within **Edge Applications** have been moved to other components as part of the migration.

> | Setting                                      | New Location      |

Author: Ajustado

  "name": "edge storage connector",
  "active": true,
  "type": "edge_storage",
  "attributes": {
    "bucket": "app-origin",
    "prefix": "/"
}'
`} />

**Creating an Edge Connector with the HTTP Connector Type**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/edge_connector/connectors \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
    "name": "example.com",
    "active": true,
    "type": "http",
    "attributes": {
      "addresses": [
        {
          "active": true,
          "address": "httpbin.org",
          "http_port": 80,
          "https_port": 443,
          "modules": null
        }
      ],
      "connection_options": {
        "dns_resolution": "preserve",
        "transport_policy": "preserve",
        "http_version_policy": "http1_1",
        "host": "httpbin.org",
        "path_prefix": "/",
        "following_redirect": true,
        "real_ip_header": "X-Real-IP",
        "real_port_header": "X-Real-PORT"
      },
      "modules": {
        "load_balancer": {
          "enabled": false,
          "config": null
        },

>         "origin_shield": {

Reviewer 3: Temos em algum lugar a explicação para os clientes que usam o Azion Origin Shield que não é a mesma coisa aqui? https://www.azion.com/en/documentation/products/secure/edge-firewall/network-layer-protection/#origin-shield

### Required action

If you have an automation that currently uses the **Edge Application - Origins** API Endpoints you must update it to use v4 endpoints.

### New Features

- **Configuration reusability**: An Edge Connector can be connected to multiple Edge Applications.
- **Configurable timeouts** `Connection Timeout` and `Read/Write Timeout` values can be configured in the Load Balancer Configuration section of the **Edge Connector**.

### Migration Benefits

- **Simplified Backend Management**: **Edge Connector** centralizes all backend connection definitions, reducing configuration complexity. Multiple **Edge Applications** can use a single **Edge Connector**, streamlining maintenance and ensuring configuration consistency.
- **Future-Ready Architecture**: **Edge Connector** provides the foundation for upcoming Azion Web Platform features and optimizations, ensuring your Edge Applications are ready to leverage new capabilities as they become available.

<LinkButton link="/en/documentation/products/secure/edge-connector" text="Visit the Edge Connector Reference"/>

---

## Custom Pages

> **Custom Pages** is an Azion feature that allows you to customize error pages for users based on the status code received from an [Edge Connector](/en/documentation/products/secure/edge-connector/) when fetching the content from your origin.

Reviewer 3: Se precisar de diferentes configurações de custom pages pra diferentes Edge connectors, cliente tem como?

        "attributes": {
          "connector": <edge_connector_id>,
          "ttl": 300,
          "uri": "/path/error_page.html",
          "custom_status_code": 100
        }
      }
    }
  ]
}'
`} />

**Binding a Custom Page to a Workload Deployment**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
      "custom_page": <custom_page_id>
    }
  }
}'
`} />

:::tip
For more information about the API parameters visit the [Custom Pages API reference](https://api.azion.com/v4#/operations/GetWorkspaceCustomPages)
:::

### What happens to my existing Error Responses?

Existing Error Responses will be automatically converted to **Custom Pages** in phases, maintaining compatibility with the original configurations. The migration will occur in waves.

### New Features

> - **Response Details**: Customize the response for an status code. Specify the URI, status code, TTL, content type, and the response body to be returned to the client.

Reviewer 2: @LuizaVSantos  O Custom Pages NÃO terá ainda a customização de content-type ou response body nesta versão. Remover.

- **Per-Status Code Edge Connector Assignment** Assign a specific **Edge Connector** to each error response page code in **Custom Pages**. This allows you to customize the response for each error code individually.

### Migration Benefits

> - **Support for Multiple Status Codes**: Configure custom responses for a wide range of HTTP status codes, including `400`, `401`, `403`, `404`, and others.

Reviewer 2: @LuizaVSantos : Multiple Status Codes já era uma feature existente, não é um beneficio da migração em si.

- **Comprehensive Configuration Options**:
  - Define a custom status code for each page.
  - Set the response Time to Live (TTL).
  - Specify the path to the custom page.
  - Link each custom page to one or more Edge Connectors.

>   - Use the Type field to associate the custom page with different contexts, such as a specific **Edge Connector**.

Reviewer 2: @LuizaVSantos Não é possível mudar o Type nesta versão.

---
title: API v4 Migration
description: >-

>   Find detailed informations about Azions plan for migration to our new API v4.

Reviewer: ```suggestion
  Find detailed information about Azions plan for migration to our new API v4.
```

**Creating a Workload**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Example Workload",
  "active": true,
  "infrastructure": 1,
  "protocols": {
    "http": {
      "versions": [
        "http1",
        "http2"
      ]
    }
  },
  "workload_domain_allow_access": true
}'
`} />

**Creating a Workload Deployment**:

<Code lang="bash" code={`curl --request POST \
  --url https://api.azion.com/v4/workspace/workloads/<workload_id>/deployments \
  --header 'Accept: application/json' \
  --header 'Authorization: Token [TOKEN VALUE]' \
  --header 'Content-Type: application/json' \
  --data '{
  "name": "Workload Deployment Example",
  "current": true,
  "active": true,
  "strategy": {
    "type": "default",
    "attributes": {
      "edge_application": <edge_application_id>
    }
  }
}'
`} />

:::tip
<<<<<<< HEAD
For more information about the API parameters visit the [Workloads API reference](https://api.azion.com/v4#/operations/GetWorkspaceWorkloads).
=======

> For more information about the API parameters visit the [Workloads API reference](https://api.azion.com/v4#/operations/GetWorkspaceWorkloads)

Reviewer: Precisa constar 2x?
